### PR TITLE
Add Video Ad Log in Dynamodb and Graphql, update User Schema, add ability to create videoAdLog

### PR DESCRIPTION
**Pull request by amaliwanag2**

### CODE REVIEW

File: graphql/database/videoAdLog/isVideoAdEligible.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import moment from 'moment'
+import VideoAdLogModel from './VideoAdLogModel'
+/**
+ * @param {Object} userContext - The user context.
+ * @param {string} userId - The user object
+ * @return {Promise<Object>}  A promise that resolves into a boolean on whether user is video ad eligible
+ */
+
+export default async (userContext, { id: userId }) =>
+  (await VideoAdLogModel.query(userContext, userId)
```
Comment (amaliwanag2) [inline graphql/database/videoAdLog/isVideoAdEligible.js:10]: if a user clicks then immedietely closes a video ad, we don't want it to count

File: graphql/database/videoAdLog/__tests__/createVideoAdLog.test.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      timestamp: '2017-05-19T13:59:46.000Z',
+      updated: '2017-05-19T13:59:46.000Z',
+      userId: 'abcdefghijklmno',
+    })
+
+    // testing nanoid
+    expect(result.id.length).toEqual(16)
+  })
+
+  it('throws if user id does not match user context false if user has  3 completed video ad logs in the last 24 hours', async () => {
```
Comment (kmjennison) [inline graphql/database/videoAdLog/__tests__/createVideoAdLog.test.js:49]: Fix test description

File: graphql/database/videoAdLog/__tests__/isVideoAdEligible.test.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+          completed: true,
+          id: '1234567890asdfg1',
+          truexEngagementId: 'asdf',
+          truexCreativeId: '1234',
+        },
+      ],
+    })
+    const result = await isVideoAdEligible(userContext, user)
+    expect(result).toEqual(false)
+  })
```
Comment (kmjennison) [inline graphql/database/videoAdLog/__tests__/isVideoAdEligible.test.js:66]: Add a test with >3 items but <3 are completed.

File: graphql/data/schema.js [hunk 4]
```diff
@@ -745,6 +754,17 @@ const widgetType = new GraphQLObjectType({
   interfaces: [nodeInterface],
 })
 
+const videoAdLogType = new GraphQLObjectType({
+  name: VIDEO_AD_LOG,
+  description: 'Vidieo Ad Log type',
```
Comment (kmjennison) [inline graphql/data/schema.js:759]: Typo

File: graphql/database/videoAdLog/VideoAdLogModel.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      timestamp: types
+        .string()
+        .isoDate()
+        .default(self.fieldDefaults.timestamp)
+        .description(`time of creation`)
+        .required(),
+      completed: types
+        .boolean()
+        .default(self.fieldDefaults.completed)
+        .description('user has aknowledged that the mission has completed'),
```
Comment (kmjennison) [inline graphql/database/videoAdLog/VideoAdLogModel.js:61]: Fix description

**Review by kmjennison — CHANGES_REQUESTED** [review]
Small changes, then LGTM and merge away!

**Pull request merged**